pallets/committee/src/lib.rs

```diff
@@ -380,48 +427,31 @@ pub mod pallet {
 
             Ok(().into())
         }
-    }
 
-    /// Initialize council members. Can only be done once.
-    /// Constituent members must be initialized later by voting by the council
-    impl<T: Config> InitializeMembers<AccountIdFor<T>> for Pallet<T> {
-        fn initialize_members(members: &[AccountIdFor<T>]) {
-            if !members.is_empty() {
-                assert_eq!(
-                    Members::<T>::iter().count(),
-                    0,
-                    "Members are already initialized!"
-                );
-                for member in members {
-                    Members::<T>::insert(member, MemberType::Council);
+        #[pallet::weight(10_000)] // TODO: Set weights
+        /// Add new constituent to the committee
+        ///
+        /// NOTE:
+        ///
+        /// This call can only be called after the approval of the committee
+        pub fn add_constituent(
+            origin: OriginFor<T>,
+            constituent: AccountIdFor<T>,
+        ) -> DispatchResultWithPostInfo {
+            T::ApprovedByCommitteeOrigin::ensure_origin(origin)?;
+
+            if let Some(ty) = <Members<T>>::get(constituent.clone()) {
+                return Err(match ty {
+                    MemberType::Council => <Error<T>>::ExistedCouncilMember,
+                    MemberType::Constituent => <Error<T>>::ExistedConstituentMember,
                 }
+                .into());
+            } else {
+                Members::<T>::insert(constituent.clone(), MemberType::Constituent);
             }
```

Comment:
We can use `try_mutate` here; if no value exist yet, we can simply replace the option, otherwise we throw based on the member (council, constituent)